docs/PortingGuide2-3.md

```diff
@@ -0,0 +1,187 @@
+
+When building OpenEXR 3.x, note that if Imath 3.x library is not found
+already installed at build time, it will be automatically downloaded and
+built as part of the OpenEXR build.
+
+
+
+Finding and using OpenEXR and Imath CMake configs
+-------------------------------------------------
+
+**OpenEXR/Imath 3.x only**
+
+If you are *only* concerned with OpenEXR/Imath 3.x going forward, this is
+the recommended way to find the libraries in a downstream project that uses
+the CMake build system:
+
+    find_package(Imath CONFIG)
+    find_package(OpenEXR CONFIG)
+
+Note that the second line may be omitted if you only need the Imath
+portions.
+
+And then your project can reference the imported targets like this:
+
+    target_link_libraries (my_target
+        PRIVATE
+            OpenEXR::OpenEXR
+            Imath::Imath
+            Imath::Half
+        )
+
+You only need the parts you use, so for example, if you only need Half and
+Imath, you can omit the OpenEXR target. Also note that in our example above,
+we have used the `PRIVATE` label, but you should specify them as `PUBLIC` if
+you are exposing those classes in your own package's public interface.
+
+
+**Accommodating OpenEXR/Imath 3.x or OpenEXR 2.x**
+
+On the other hand, to accommodate both 2.x and 3.x, it's admittedly
+inconvenient because the packages and the import targets have changed their
+names.  We have found the following idioms to work:
+
+Finding either/both packages:
+
+    # First, try to find just the right config files
+    find_package(Imath CONFIG)
+    if (NOT TARGET Imath::Imath)
+        # Couldn't find Imath::Imath, maybe it's older and has IlmBase?
+        find_package(IlmBase CONFIG)
+    endif ()
+    find_package(OpenEXR CONFIG)
+
+
+To link against them, we use CMake generator expressions so that we can
+referencing *both* sets of targets, but it will only use the ones
+corresponding to the package version that was found.
+
+    target_link_libraries (my_target
+        PRIVATE
+            # For OpenEXR/Imath 3.x:
+              $<$<TARGET_EXISTS:OpenEXR::OpenEXR>:OpenEXR::OpenEXR>
+              $<$<TARGET_EXISTS:Imath::Imath>:Imath::Imath>
+              $<$<TARGET_EXISTS:Imath::Half>:Imath::Half>
+            # For OpenEXR 2.4/2.5:
+              $<$<TARGET_EXISTS:OpenEXR::IlmImf>:OpenEXR::IlmImf>
+              $<$<TARGET_EXISTS:IlmBase::Imath>:IlmBase::Imath>
+              $<$<TARGET_EXISTS:IlmBase::Half>:IlmBase::Half>
+              $<$<TARGET_EXISTS:IlmBase::IlmThread>:IlmBase::IlmThread>
+              $<$<TARGET_EXISTS:IlmBase::Iex>:IlmBase::Iex>
+        )
+
+Again, you can eliminate the references to any of the individual libaries
+that you don't actually need for your application.
+
+
+Imath include files are in a different subdirectory
+---------------------------------------------------
+
+Imath 3.0 will copy its headers to some `include/Imath` subdirectory
+instead of the old `include/OpenEXR`.
```

Comment:
Currently, people are possibly tripped up by PyImath being published into OpenEXR. Probably we need to confirm that's happening and revise the behavior

Reply:
I, myself, have actually never used PyImath and am not sure I can even articulate its purpose (I would use numpy for my linear algebra needs in Python), so I don't have any expertise or opinions on how we should package it or even know how 3.x differs from 2.x. Somebody with a dog in that race ought to make final packaging decisions and explain how it should be consumed.

Reply:
This is fixed by #88, the stray OpenEXR in PyImath was an oversight. PyImath is used by Alembic.